mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixtureModel.scala

```diff
@@ -83,5 +95,81 @@ class GaussianMixtureModel(
       p(i) /= pSum
     }
     p
-  }  
+  }
+}
+
+@Experimental
+object GaussianMixtureModel extends Loader[GaussianMixtureModel] {
+
+  private object SaveLoadV1_0 {
+
+    case class Data(weight: Double, mu: Vector, sigma: Matrix)
+
+    def formatVersionV1_0 = "1.0"
+
+    def classNameV1_0 = "org.apache.spark.mllib.clustering.GaussianMixtureModel"
+
+    def save(
+        sc: SparkContext,
+        path: String,
+        weights: Array[Double],
+        gaussians: Array[MultivariateGaussian]): Unit = {
+
+      val sqlContext = new SQLContext(sc)
+      import sqlContext.implicits._
+
+      // Create JSON metadata.
+      val metadata = compact(render
+        (("class" -> classNameV1_0) ~ ("version" -> formatVersionV1_0) ~ ("k" -> weights.length)))
+      sc.parallelize(Seq(metadata), 1).saveAsTextFile(Loader.metadataPath(path))
+
+      // Create Parquet data.
+      val dataArray = Array.tabulate(weights.length){ i =>
+        Data(weights(i), gaussians(i).mu, gaussians(i).sigma)
+      }
+      val dataRDD: DataFrame = sc.parallelize(dataArray, 1).toDF()
+      dataRDD.saveAsParquetFile(Loader.dataPath(path))
+    }
+
+    def load(sc: SparkContext, path: String) : GaussianMixtureModel = {
+      val datapath = Loader.dataPath(path)
+      val sqlContext = new SQLContext(sc)
+      val dataRDD = sqlContext.parquetFile(datapath)
+      val numGaussians = dataRDD.count().toInt
+      val dataArray = dataRDD.select("weight", "mu", "sigma").collect()
+      // Check schema explicitly since erasure makes it hard to use match-case for checking.
+      Loader.checkSchema[Data](dataRDD.schema)
+
+      val (weights, gaussians) = Array.tabulate(numGaussians) { i =>
+        val currentMu = dataArray(i).getAs[Vector](1)
+        val currentSigma = dataArray(i).getAs[Matrix](2)
+        (dataArray(i).getAs[Double](0), new MultivariateGaussian(currentMu, currentSigma))
```

Comment:
`getAs[Double]` -> `getDouble`. Actually, it is better to write

~~~scala
val (weights, gaussians) = dataArray.map { case Row(weight: Double, mu: Vector, sigma: Matrix) =>
  (weight, new MultivariateGaussian(mu, sigma))   
}.unzip
~~~